diesel/src/sqlite/connection/mod.rs

```diff
@@ -67,17 +67,19 @@ impl Connection for SqliteConnection {
         Self::Backend: HasSqlType<T::SqlType>,
         U: Queryable<T::SqlType, Self::Backend>,
     {
-        let statement = try!(self.prepare_query(&source.as_query()));
-        let mut statement_ref = statement.borrow_mut();
-        StatementIterator::new(&mut statement_ref).collect()
+        let mut statement = try!(self.prepare_query(&source.as_query()));
+        let statement_use = StatementUse::new(&mut statement);
+        let iter = StatementIterator::new(statement_use);
+        iter.collect()
```

Comment:
I assume Rust doen't let you to continue to use `StatementIterator::new(statement_use).collect()` because temporary vars with too short a scope?